Addition of Docker Image and User Example

gamedaygeorge opened the pull request:
This PR proposes to:

- include a Docker Image folder to help with deployment locally in Visual Studio Code, as well as using GitHub Code Spaces; and
- add a user example notebook, including a user examples folder.

Works have been conducted as a part of writing an article (for Earth Online) which we hope to link to the new user example notebook.

gamedaygeorge commented:
@libbyrose FYI on this PR

TonioF commented:
Thank you for the pr. One of our next steps would have been anyway to provide a docker image, so your contribution comes at a good time. As I have seen you were adding commits after you posted the pr, I wanted to ask: Is it ready for review or are you still working on it?

gamedaygeorge commented:
As you saw, I am working on a few tweaks to the documentation. I am not expert on Docker Containers, but what I have works for running in GitHub Codespace. I will upload the new docs next week, and then maybe you can have a look?

gamedaygeorge commented:
@TonioF I have committed the latest on our fork, so this should be ready to take a look at now.

TonioF commented on notebooks/external_user_examples/README.md:
```diff
@@ -0,0 +1,24 @@
+
+![Unsupported](https://img.shields.io/badge/status-unsupported-red)
+
+Welcome to the **external_user_examples** folder!
```
```suggestion
Welcome to the **user_examples** folder!
```

TonioF reviewed (changes requested):
Apologies for the delay: We now have reviewed your pull request and found that it is actually a very good addition to the toolbox. Thank you very much for your efforts. I just found minor things to address. Also, could you please rename the folder "external_user_examples" to just "user_examples", please?
Finally, I found that the last cell of the user examples notebook did not produce a visualisation for me. This will not be a reason for me not to accept and merge this pr, but maybe you want to check this.

gamedaygeorge commented:
I have progressed the changes on this PR here today, in summary:

- Accepted the proposed changes
- Update the name of the user_examples folder
- Fixed the issue with the plot in the last cell of the CCI-SeaIce-Animation.ipynb [here](https://github.com/gamedaygeorge/esa-climate-toolbox/blob/main/notebooks/user_examples/CCI-SeaIce-Animation.ipynb) not generating a plot
- Adjusted the dockerfile and devcontainer configuration
- Updated documentation and screenshots

This should be reflected in the PR.

gamedaygeorge commented:
Great - thanks @TonioF !
